spinnaker/spinnaker_system/google_http_lb_upsert_test.py

```diff
@@ -0,0 +1,894 @@
+        bindings['GCE_CREDENTIALS'],
+        bindings['TEST_GCE_REGION'],
+      ),
+      application=self.TEST_APP
+    )
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    (contract_builder.new_clause_builder('Health Check Removed', retryable_for_secs=30)
+     .list_resource('httpHealthChecks')
+     .excludes_path_value('name', self.__proto_hc['name'])
+    )
+    (contract_builder.new_clause_builder('Url Map Removed',
+                                         retryable_for_secs=30)
+     .list_resource('urlMaps')
+     .excludes_path_value('name', self.__lb_name)
+    )
+    (contract_builder.new_clause_builder('Forwarding Rule Removed',
+                                         retryable_for_secs=30)
+     .list_resource('globalForwardingRules')
+     .excludes_path_value('name', self.__lb_name)
+    )
+
+    return st.OperationContract(
+      self.new_post_operation(
+        title='delete_load_balancer', data=payload, path='tasks'),
+      contract=contract_builder.build())
+
+
+  def change_health_check(self):
+    '''Changes the health check associated with the LB.
+    '''
+    upsert = copy.deepcopy(self.__proto_upsert)
+    hc = copy.deepcopy(self.__proto_hc)
+    hc['requestPath'] = '/hello'
+    hc['port'] = 8080
+    self._set_all_hcs(upsert, hc)
+
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[upsert],
+      descriptiopn='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    hc_link = self._make_hc_link(hc['name'])
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._make_contract_clauses(contract_builder, upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='change health checks',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def change_backend_service(self):
+    '''Changes the default backend service associated with the LB.
+    '''
+    hc = copy.deepcopy(self.__proto_hc)
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
+    hc['name'] = 'updated-hc'
+    hc['requestPath'] = '/hello'
+    hc['port'] = 8080
+
+    bs_upsert['defaultService']['healthCheck'] = hc
+    payload = self.agent.make_json_payload_from_kwargs(
+      job=[bs_upsert],
+      descriptiopn='Upsert L7 Load Balancer: ' + self.__lb_name,
+      application=self.TEST_APP
+    )
+
+    hc_link = self._make_hc_link(hc['name'])
+    proto_hc_link = self._make_hc_link(self.__proto_hc['name'])
+
+    contract_builder = gcp.GcpContractBuilder(self.gcp_observer)
+    self._make_contract_clauses(contract_builder, bs_upsert)
+
+    return st.OperationContract(
+      self.new_post_operation(title='change backend services',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+  def add_host_rule(self):
+    '''Adds a host rule to the url map.
+    '''
+    bs_upsert = copy.deepcopy(self.__proto_upsert)
```

Comment:
This would flow better moving #442 after #445

As a rule of thumb, I like to try to keep locality of reference. You create an object, then prepare an attribute value, then add that new attribute to the object you'd created before. If you change the order, you prepare a value, then create an object and set the value on it. The difference is subtle, but magnified a bit more in that you then go on to use that object and forget about the value rather than coming back to it.